characterEntities/Entity.java

```diff
@@ -290,7 +286,9 @@ public void update() {
             stunCounter = 0;
         }
 
-        if (currentAbilityAnimation != null && currentAbilityAnimation.isDone()) {
+        if (currentAbilityAnimation != null && !currentAbilityAnimation.isDone()) {
```

Comment:
write this:
if (currentAbilityAnimation != null) {
    currentAbilityAnimatiaon.update();
    if (currentAbilityAnimation.isDone()) {
        //rest of the code
    }
}
this is so that if the animation finishes, the entity can immediately return to neutral state